esphome/components/dht/dht.cpp

```diff
@@ -24,6 +24,12 @@ void DHT::dump_config() {
     ESP_LOGCONFIG(TAG, "  Model: DHT22 (or equivalent)");
   }
 
+  if (this->use_internal_pullup_) {
+    ESP_LOGCONFIG(TAG, "  Internal pull-up resistor enabled.");
+  } else {
+    ESP_LOGCONFIG(TAG, "  Internal pull-up resistor disabled.");
+  }
```

Comment:
```suggestion
    ESP_LOGCONFIG(TAG, "  Internal pull-up resistor: %s", ONOFF(this->use_internal_pullup_));
```

Use one of these macros:
https://github.com/esphome/esphome/blob/f7f8bf4da44b83796a301bc486d77fdb46d8ed82/esphome/core/log.h#L161-L163